packages/@o3r/design/src/core/design-token/parsers/design-token-parser.interface.ts

```diff
@@ -0,0 +1,74 @@
+import type { DesignToken, DesignTokenExtensions, DesignTokenGroup, DesignTokenGroupExtensions } from '../design-token-specification.interface';
+
+export interface ParentReference {
+  /** Design Token name */
+  name: string;
+  /** Design Token Group node */
+  tokenNode: DesignTokenGroup;
+}
+
+/**
+ * Function rendering the Design Token Value
+ * @param tokenStructure Parsed Design Token
+ * @param variableSet Complete list of the parsed Design Token
+ */
+// eslint-disable-next-line no-use-before-define
+export type TokenValueRenderer = (tokenStructure: DesignTokenVariableStructure, variableSet: Map<string, DesignTokenVariableStructure>) => string;
+
+/**
+ * Function rendering the Design Token Key
+ * @param tokenStructure Parsed Design Token
+ */
+// eslint-disable-next-line no-use-before-define
+export type TokenKeyRenderer = (tokenStructure: DesignTokenVariableStructure) => string;
+
+/** Complete list of the parsed Design Token */
+// eslint-disable-next-line no-use-before-define
+export type DesignTokenVariableSet = Map<string, DesignTokenVariableStructure>;
+
+/** Parsed Design Token variable */
+export interface DesignTokenVariableStructure {
+  /** Design Token Extension */
+  extensions: DesignTokenGroupExtensions & DesignTokenExtensions;
```

Comment:
What is the & for ? One interface extends the other.

Reply:
Actually there are 2 type of extensions: the one set at node level only, the one set at token level only.
As the ones set as node levels will affect the child token, the parser it actually merging the 2 types in a single object following the override prioritize.